New Resource: `azurerm_application_load_balancer_frontend`

ziyeqf opened the pull request.

JackStromberg commented on website/docs/r/service_networking_application_load_balancer_frontend.html.markdown:
```diff
@@ -0,0 +1,66 @@
+
+* `name` - (Required) The name which should be used for this Application Gateway for Containers Frontend. Changing this forces a new resource to be created.
+
+* `location` - (Required) The Azure Region where the Application Gateway for Containers Frontend should exist. Changing this forces a new resource to be created. Available Regions could be found in [here](https://learn.microsoft.com/en-us/azure/application-gateway/for-containers/overview#supported-regions).
```
The word region can be lowercase.  In the last sentence, I'd change could to can and remove the word in.

Available regions can be found [here](link).

ms-zhenhua committed:
update api version

tombuildsstuff committed:
Merge pull request #23178 from sinbai/redisenterprise/upgrade_api_version

`redisenterprise` : upgrade SDK to v2023-07-01

tombuildsstuff committed:
updating to include #23178

tombuildsstuff committed:
Merge pull request #23185 from hashicorp/redisenterprise/test-names

redisenterprise: updating the test names to include `TestAcc` since that's used to run the tests

Commit:
Updating based on f72f7d923

tombuildsstuff committed:
Merge pull request #23189 from hashicorp/auto-pr/f72f7d923

Auto PR: Regenerating based on (42c000ff1ed9581abcf7eaf6f392bd3088fbb57b)

tombuildsstuff committed:
dependencies: updating to `v0.20230906.1160501` of `github.com/hashicorp/go-azure-sdk`

tombuildsstuff committed:
refactor: updating to account for the renamed signatures

paulomarquesc committed:
New Resource/Data Source: `netapp_volume_quota_rule` (#23042)

mbfrahry committed:
Update CHANGELOG.md for #23042

sinbai committed:
support new region

aristosvo committed:
`azurerm_kubernetes_cluster` - Update Azure CNI to Azure CNI Overlay enabled (#22709)

* `azurerm_kubernetes_cluster` - Update Azure CNI to Azure CNI Overlay enabled

* fmt

* Test data naming improved

* Validate reverse overlay

* Make the upgrade one-way

* Update kubernetes_cluster_resource.go

Co-authored-by: stephybun <steph@hashicorp.com>

---------

Co-authored-by: stephybun <steph@hashicorp.com>

stephybun committed:
Update for #23146 #23124 #23096 #22906 #23160 #23106 #22709

tombuildsstuff committed:
Merge pull request #23191 from hashicorp/dependencies/go-azure-sdk

dependencies: updating to v0.20230906.1160501 of `github.com/hashicorp/go-azure-sdk`

tombuildsstuff committed:
updating to include #23191

tombuildsstuff committed:
Merge pull request #23200 from sinbai/redisenterprise/support_new_region

`azurerm_redis_enterprise_cluster` : support for new location `Brazil South`

tombuildsstuff committed:
updating to include #23200

Commit:
Updating based on 4d4c013f7

jackofallops committed:
add note for app_setting replacement value

teowa committed:
fix webhook.service_uri (#23203)

jackofallops committed:
`appservice` - add note for app_setting replacement value (#23206)

`appservice` - add note for app_setting replacement value

tombuildsstuff committed:
Merge pull request #23205 from hashicorp/auto-pr/4d4c013f7

Auto PR: Regenerating based on (c697b9f9ea42284117bddc57d84398b7a1b93d93)

jackofallops committed:
add schemas and structs for scale rules types

sinbai committed:
`azurerm_api_management_api_policy`, `azurerm_api_management_api_operation_policy`, `azurerm_api_management_product_policy` : fix parsing resource id error (#23128)

* fix parsing id error

* update code

stephybun committed:
Update for #23128

teowa committed:
`azurerm_key_vault_certificate` - point to new version when `certificate` is updated (#23135)

* point to new version when import updated

* update secret doc

* fix doc char case

catriona-m committed:
Update CHANGELOG.md #23135

ziyeqf committed:
`azurerm_site_recovery_replicated_vm`: fix `network_interface`, use the default hash func (#23199)

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

catriona-m committed:
Update CHANGELOG.md #23199

wuxu92 committed:
skip wait status when creating certification with Unknown Issuer

stephybun committed:
`azurerm_mssql_database` - split create and update methods (#23209)

mbfrahry committed:
Update CHANGELOG.md for #23209

liuwuliuyun committed:
`azurerm_key_vault_managed_storage_account` - fix TestAccKeyVaultManagedStorageAccount_* (#23022)

mbfrahry committed:
Update CHANGELOG.md for #23022

mbfrahry committed:
v3.72.0

SebSa committed:
`azurerm_maps_account` - Support for `local_authentication_enabled` (#23216)

mbfrahry committed:
Update CHANGELOG.md for #23216

wuxu92 committed:
fix dereference may cause panic in apim

SebSa committed:
`azurerm_iothub`: Support for `disableLocalAuth` / `local_authentication_enabled` (#23119)

* add local_authentication_enabled to iothub resource

* apply suggested fixes

* terrafmt

teowa committed:
`azurerm_application_gateway` - support for `verify_client_certificate_revocation` in `ssl_profile` block (#23181)

* ssl_profile.0.verify_client_cert_revocation

* add doc

* remove cert abbr

* fix test

* send None by default

tombuildsstuff committed:
dependencies: updating

```
 $ go get -u
go: downloading github.com/hashicorp/terraform-plugin-sdk/v2 v2.29.0
go: downloading github.com/hashicorp/go-azure-sdk v0.20230907.1113401
go: downloading github.com/google/uuid v1.3.1
go: downloading golang.org/x/net v0.15.0
go: downloading golang.org/x/crypto v0.13.0
go: downloading github.com/rickb777/date v1.20.2
go: downloading github.com/hashicorp/terraform-plugin-testing v1.5.1
go: downloading github.com/hashicorp/terraform-plugin-go v0.19.0
go: downloading github.com/hashicorp/go-plugin v1.5.1
go: downloading golang.org/x/oauth2 v0.12.0
go: downloading software.sslmate.com/src/go-pkcs12 v0.2.1
go: downloading github.com/hashicorp/terraform-json v0.17.1
go: downloading github.com/hashicorp/terraform-exec v0.19.0
go: downloading golang.org/x/text v0.13.0
go: downloading github.com/fatih/color v1.15.0
go: downloading github.com/mattn/go-isatty v0.0.19
go: downloading google.golang.org/grpc v1.58.0
go: downloading golang.org/x/sys v0.12.0
go: downloading github.com/zclconf/go-cty v1.14.0
go: downloading google.golang.org/appengine v1.6.8
go: downloading github.com/hashicorp/hcl/v2 v2.18.0
go: downloading github.com/hashicorp/hc-install v0.6.0
go: downloading google.golang.org/protobuf v1.31.0
go: downloading google.golang.org/genproto/googleapis/rpc v0.0.0-20230822172742-b8732ec3820d
go: downloading google.golang.org/genproto v0.0.0-20230822172742-b8732ec3820d
go: downloading github.com/hashicorp/terraform-registry-address v0.2.2
go: downloading github.com/hashicorp/terraform-svchost v0.1.1
go: downloading google.golang.org/genproto v0.0.0-20230803162519-f966b187b2e5
go: downloading github.com/apparentlymart/go-textseg/v15 v15.0.0
go: downloading github.com/ProtonMail/go-crypto v0.0.0-20230717121422-5aa5874ade95
go: downloading github.com/ProtonMail/go-crypto v0.0.0-20230828082145-3c4c8a2d2371
go: downloading golang.org/x/exp v0.0.0-20230809150735-7b3493d9a819
go: downloading golang.org/x/exp v0.0.0-20230905200255-921286631fa9
go: downloading github.com/Microsoft/go-winio v0.6.1
go: downloading github.com/go-git/gcfg v1.5.1-0.20230307220236-3a3c6141e376
go: downloading github.com/go-git/go-git/v5 v5.8.1
go: downloading github.com/imdario/mergo v0.3.15
go: downloading github.com/jhump/protoreflect v1.15.1
go: downloading golang.org/x/sync v0.3.0
go: downloading golang.org/x/term v0.12.0
go: downloading golang.org/x/tools v0.13.0
go: upgraded github.com/ProtonMail/go-crypto v0.0.0-20210428141323-04723f9f07d7 => v0.0.0-20230828082145-3c4c8a2d2371
go: added github.com/apparentlymart/go-textseg/v15 v15.0.0
go: upgraded github.com/btubbs/datetime v0.1.0 => v0.1.1
go: added github.com/cloudflare/circl v1.3.3
go: upgraded github.com/fatih/color v1.13.0 => v1.15.0
go: upgraded github.com/golang/protobuf v1.5.2 => v1.5.3
go: upgraded github.com/google/uuid v1.3.0 => v1.3.1
go: upgraded github.com/hashicorp/go-azure-sdk v0.20230906.1160501 => v0.20230907.1113401
go: upgraded github.com/hashicorp/go-hclog v1.4.0 => v1.5.0
go: upgraded github.com/hashicorp/go-plugin v1.4.8 => v1.5.1
go: upgraded github.com/hashicorp/hc-install v0.5.0 => v0.6.0
go: upgraded github.com/hashicorp/hcl/v2 v2.16.2 => v2.18.0
go: upgraded github.com/hashicorp/terraform-exec v0.18.1 => v0.19.0
go: upgraded github.com/hashicorp/terraform-json v0.16.0 => v0.17.1
go: upgraded github.com/hashicorp/terraform-plugin-go v0.14.3 => v0.19.0
go: upgraded github.com/hashicorp/terraform-plugin-log v0.8.0 => v0.9.0
go: upgraded github.com/hashicorp/terraform-plugin-sdk/v2 v2.26.1 => v2.29.0
go: upgraded github.com/hashicorp/terraform-plugin-testing v1.0.0 => v1.5.1
go: upgraded github.com/hashicorp/terraform-registry-address v0.1.0 => v0.2.2
go: upgraded github.com/hashicorp/terraform-svchost v0.0.0-20200729002733-f050f53b9734 => v0.1.1
go: upgraded github.com/hashicorp/yamux v0.0.0-20210316155119-a95892c5f864 => v0.1.1
go: upgraded github.com/mattn/go-colorable v0.1.12 => v0.1.13
go: upgraded github.com/mattn/go-isatty v0.0.14 => v0.0.19
go: upgraded github.com/rickb777/date v1.12.5-0.20200422084442-6300e543c4d9 => v1.20.2
go: upgraded github.com/rickb777/plural v1.2.0 => v1.4.1
go: upgraded github.com/vmihailenco/tagparser v0.1.1 => v0.1.2
go: upgraded github.com/zclconf/go-cty v1.13.1 => v1.14.0
go: upgraded golang.org/x/crypto v0.9.0 => v0.13.0
go: added golang.org/x/exp v0.0.0-20230905200255-921286631fa9
go: upgraded golang.org/x/mod v0.8.0 => v0.12.0
go: upgraded golang.org/x/net v0.10.0 => v0.15.0
go: upgraded golang.org/x/oauth2 v0.4.0 => v0.12.0
go: upgraded golang.org/x/sys v0.8.0 => v0.12.0
go: upgraded golang.org/x/text v0.9.0 => v0.13.0
go: upgraded golang.org/x/tools v0.6.0 => v0.13.0
go: upgraded google.golang.org/appengine v1.6.7 => v1.6.8
go: upgraded google.golang.org/genproto v0.0.0-20230110181048-76db0878b65f => v0.0.0-20230803162519-f966b187b2e5
go: added google.golang.org/genproto/googleapis/rpc v0.0.0-20230822172742-b8732ec3820d
go: upgraded google.golang.org/grpc v1.53.0 => v1.58.0
go: upgraded google.golang.org/protobuf v1.28.1 => v1.31.0
go: upgraded software.sslmate.com/src/go-pkcs12 v0.2.0 => v0.2.1
```

neil-yechenwei committed:
New Resource: `azurerm_virtual_hub_routing_intent` (#23138)

* New Resource: azurerm_virtual_hub_routing_intent

* update code

* update code

catriona-m committed:
Update CHANGELOG.md #23138

myc2h6o committed:
New resource `azurerm_iothub_endpoint_cosmosdb_account` (#23065)

* New resource `azurerm_iothub_endpoint_cosmosdb_account`

* resolve comments

catriona-m committed:
Update CHANGELOG.md #23065

tombuildsstuff committed:
pluginsdk/resource: updating the import for `resource.StateChangeConf`

tombuildsstuff committed:
pluginsdk/resource: updating the reference to `StateRefreshFunc` to use `retry.StateRefreshFunc`

tombuildsstuff committed:
pluginsdk/retries: updating `RetryFunc` to reference `resource.RetryFunc`

tombuildsstuff committed:
pluginsdk/retries: updating `RetryError` to reference `resource.RetryError`

tombuildsstuff committed:
acceptance/ssh: updating the retry func

tombuildsstuff committed:
acceptance/plugin_sdk_aliases: updating a reference to `retry.StateChangeConf`

wuxu92 committed:
no need check map nil

wuxu92 committed:
restore lint comment

ms-henglu committed:
`azurerm_kubernetes_cluster` - update version in test cases

tombuildsstuff committed:
Merge pull request #23221 from hashicorp/dependencies/update

dependencies: updating

tombuildsstuff committed:
updating to include #23221

tombuildsstuff committed:
Merge pull request #23230 from ms-henglu/branch-230911-update-aks-kubernetes-version

`azurerm_kubernetes_cluster` - update version in test cases

tombuildsstuff committed:
Merge pull request #23219 from wuxu92/fix/apimpanic

`azurerm_api_management_backend`: fix dereference pointer may cause panic

tombuildsstuff committed:
updating to include #23219

ms-zhenhua committed:
go mod vendor

tombuildsstuff committed:
Merge pull request #23158 from sinbai/securitycenter/migrate_pandora_sdk_for_assessmentmetadata

`azurerm_security_center_assessment_policy` : refactoring to use hashicorp/go-azure-sdk

tombuildsstuff committed:
updating to include #23158

tombuildsstuff committed:
go mod vendor / fixing the dependencies

tombuildsstuff committed:
Merge pull request #23235 from hashicorp/b/fixing-deps

go mod vendor / fixing the dependencies

tombuildsstuff committed:
r/security_center_assessment_policy: fixing a compile time issue

This should have been included in the last PR, but was an oversight (I'd missed this wasn't just a vendor issue)

Commit:
Updating based on 18b7a804d

tombuildsstuff committed:
Merge pull request #23236 from hashicorp/auto-pr/18b7a804d

Auto PR: Regenerating based on (929dbb69b1a44139157ce606784fc307f4f65414)

tombuildsstuff committed:
Merge pull request #23233 from ms-zhenhua/upgrade-compute-disk-api

`azurerm_managed_disk` - upgrade API from `2022-03-02` to `2023-04-02`

tombuildsstuff committed:
updating to include #23233

ms-zhenhua committed:
add code

neil-yechenwei committed:
update code

UppyAU committed:
azurerm_web_application_firewall_policy add enabled / disabled state for Custom Rules (#23163)

* Added the Enable Field for State on Custom Rules

* TestAccWebApplicationFirewallPolicy_complete Updated

* Fix linting on Enable in test

* Created a Test involving Custom Rules Enable State

* Corrected Go Path in ENV Vars and Ran terrafmt

stephybun committed:
Update for #23163

xuzhang3 committed:
`azurerm_redis_cache` - Check if `redis_configuration.aof_backup_enabled` configured  (#22774)

* add check for aof_backup_enabled

* update CRUD timeout

* reset crud timeout

* ignore aof_backup_enabled if SKU is not premium

* fix lint

---------

Co-authored-by: xuzhang3 <Zhangxu894765>

stephybun committed:
Update for #22774

tombuildsstuff committed:
Merge pull request #23122 from neil-yechenwei/msteamschannel02

`azurerm_bot_channel_ms_teams` - support for `deployment_environment`

tombuildsstuff committed:
updating to include #23122

mbfrahry committed:
Upgrade to v0.61.0 of go-azure-helpers

ms-zhenhua committed:
resolve comments 0913

sinbai committed:
fix issue 23187

tombuildsstuff committed:
Merge pull request #23249 from hashicorp/go-azure-helpers-62

dependencies - updating to `v0.61.0` of `github.com/hashicorp/go-azure-helpers`

tombuildsstuff committed:
updating to include #23249

stephybun committed:
contributor docs: update naming rules on bools and improve formatting (#23254)

* update naming rules on bools and improve formatting

* add spacing to ease reading of raw markdown

* remove "in" in rename example

Co-authored-by: Tom Harvey <tombuildsstuff@users.noreply.github.com>

* added point on the naming of time properties

---------

Co-authored-by: Tom Harvey <tombuildsstuff@users.noreply.github.com>

tombuildsstuff committed:
Merge pull request #23241 from ms-zhenhua/managed-disk-enhancement

`azurerm_managed_disk` - Support `optimized_frequent_attach_enabled` and `performance_plus_enabled`

tombuildsstuff committed:
updating to include #23241

tombuildsstuff committed:
Merge pull request #23256 from sinbai/mssql/fix_issue_23187

`azurerm_mssql_elasticpool` : fix `Hyperscale` tier `license_type` error

tombuildsstuff committed:
updating to include #23256

tombuildsstuff committed:
Merge pull request #23214 from wuxu92/fix/unknowCert

`azurerm_key_vault_certificate`: skip wait status when creating certification with Unknown Issuer

tombuildsstuff committed:
updating to include #23214

jackofallops committed:
add scale rules with auth

mbfrahry committed:
`azurerm_api_management` - split create and update methods  (#23259)

mbfrahry committed:
Update CHANGELOG.md for #23259

katbyte committed:
v3.73.0

sinbai committed:
fix acc test failure

ms-zhenhua committed:
New Resource: `azurerm_new_relic_tag_rule` (#23062)

* add code

* add code

* add code

* add code

* fix go vendor

* add code

* resolve comments 0621

* go mod vendor

* resolve comments 0828

* resolve comments 0915

markrzasa committed:
add identity block to eventgrid domain data source (#23274)

* add identity to eventgrid domain data source

* fix terraform format errors

jackofallops committed:
rework to cater for multiple auth blocks

jackofallops committed:
add docs update for scale rules

tombuildsstuff committed:
Merge pull request #23276 from sinbai/mssql/fix_acc_test_failure

`azurerm_mssql_database_extended_auditing_policy` : fix acc test failure

mbfrahry committed:
dependencies - swap Virtual Machine Scale Set Ids for `hashicorp/go-azure-helpers/commonids` (#23272)

mbfrahry committed:
Update CHANGELOG.md for #23272

nerddtvg committed:
Add properties to error message (#23271)

lonegunmanb committed:
amend missing data source attributes (#23279)

neil-yechenwei committed:
`azurerm_postgresql_flexible_server` - support new value `4193280` for `storage_mb` (#23277)

* azurerm_postgresql_flexible_server - support new value 4193280 for storage_mb

* update code

hashicorp-tsccr[bot] committed:
Result of tsccr-helper -log-level=info -pin-all-workflows .

teowa committed:
`azurerm_monitor_diagnostic_setting` - `retention_policy` deprecated in favor of `azurerm_storage_management_policy` (#23260)

* enhance deprecation message

* add more copy

magodo committed:
`azurerm_storage_account` - `azure_file_authentication.0.active_directory` supports setting `domain_name` and `domain_guid` when `directory_type` is `AADKERB` (#22833)

* `azurerm_storage_account` - `azure_file_authentication.0.active_directory` supports setting `domain_name` and `domain_guid` when `directory_type` is `AADKERB`

* Update test

* Update

* Update

* Update doc

* terrafmt

* Avoid O+C

* fmt

* Always expand the ad block

tombuildsstuff committed:
Merge pull request #23293 from hashicorp/tsccr-auto-pinning/trusted/2023-09-18

SEC-090: Automated trusted workflow pinning (2023-09-18)

tombuildsstuff committed:
helpers/validate: adding a test for .net's `TimeSpan.MaxValue` used by ServiceBus

tombuildsstuff committed:
dependencies: downgrading to `v1.12.5` of `github.com/rickb777/date`

magodo committed:
`azurerm_storage_account` - For `Storage` kind account, avoid setting some unsupported blob properties (#23288)

* `azurerm_storage_account` - For `Storage` kind account, avoid setting some unsupported blob properties

* Explicitly mention Storage v1

tombuildsstuff committed:
Merge pull request #23296 from hashicorp/b/iso8601-dateparsing

dependencies: downgrading to `v1.12.5` of `github.com/rickb777/date`

tombuildsstuff committed:
updating to include #23296

tombuildsstuff committed:
Removing #23272 from the changelog

This is an internal refactor so can be omitted as it's not user-facing

jackofallops committed:
add validation for concurrent_requests

ms-zhenhua committed:
`azurerm_shared_image_version` - Support `replicated_region_deletion_enabled` and `target_region.exclude_from_latest_enabled` (#23147)

* add code

* resolve comments 0905

teowa committed:
`azurerm_monitor_scheduled_query_rules_alert_v2` - add doc for `metric_measure_column` limitation (#23300)

* add doc

* fix

jackofallops committed:
Merge pull request #23294 from hashicorp/f/containerapp-scale-rules

jackofallops committed:
Update CHANGELOG.md for #23294

sinbai committed:
`azurerm_api_management_api` : fix service is unavailable while updating `azurerm_api_management_api`  (#23011)

* fix issue 22973

* update code

jackofallops committed:
fix docker update settings and add test coverage

ms-zhenhua committed:
`azurerm_network_interface` - Support `auxiliary_mode`, `auxiliary_sku` (#22979)

* add code

* resolve comments 0817

* resolve comments 0818

* resolve comments 0906

* resolve comments 0914

jackofallops committed:
Merge pull request #23303 from hashicorp/b/app-service-docker-app-stack-update

jackofallops committed:
Update CHANGELOG.md for #23303

ziyeqf committed:
`azurerm_recovery_services_vault_resource_guard_association`: fix acctest (#23301)

* `azurerm_recovery_services_vault_resource_guard_association`: fix acctest

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

* add comments

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

---------

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

ziyeqf committed:
`azurerm_app_configuration`: fix a potential crash (#23302)

* bugfix #23290

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

* update per comment

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

---------

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

lonegunmanb committed:
`azurerm_linux_web_app_slot` `azurerm_linux_web_app` -  Add length check to avoid potential index out of range panic (#21328)

jackofallops committed:
Update CHANGELOG.md for #21328

neil-yechenwei committed:
`azurerm_bot_channel_web_chat` - `site_names` is removed in favour of `site` (#23161)

* azurerm_bot_channel_web_chat - support for site

* update code

* update code

* update code

* update code

* update code

* update code

* update code

* update code

* update code

* update code

* update code

* update code

* update code

* update code

* update code

* update code

* update code

ziyeqf committed:
DataSource:`azurerm_container_app_environment`: fix `log_analytics_workspace_name` (#23298)

jackofallops committed:
Update CHANGELOG.md for #23298

csimpf committed:
Update function_app_host_keys.html.markdown - fix attributes section (#23307)

Align wording with other documentation

sinbai committed:
`azurerm_mssql_elasticpool` : fix `license_type` can't be set to null when changing from vCore to DTU model issue (#23262)

* fix issue 22311

* update code

* update code

teowa committed:
app gateway doc: add v1 sku deprecation (#23310)

teowa committed:
fix update (#23263)

TheisFerre committed:
changed sdk version to 2022-01-01 for mysql flexible server

neil-yechenwei committed:
`azurerm_bot_channel_directline` - support for `user_upload_enabled`, `endpoint_parameters_enabled` and `storage_enabled` (#23149)

`azurerm_bot_channel_directline` - support for `user_upload_enabled`, `endpoint_parameters_enabled` and `storage_enabled`

magodo committed:
Update CHANGELOG.md for #23149

sinbai committed:
New Resource : `azurerm_arc_resource_bridge_appliance` (#23108)

* support new resource for resource connector RP

* run make generate

* update code

* update code

* remove version

* update code

* update code

jackofallops committed:
add zone_redundant_enabled property and tests (#23313)

Co-authored-by: matt-lethargic <mattlethargic@gmail.com>

jiaweitao001 committed:
New resource: `azurerm_function_app_connection` (#23127)

jackofallops committed:
Update CHANGELOG.md for #23127

mbfrahry committed:
dependencies - swap Virtual Machine Ids for `hashicorp/go-azure-helpers/commonids` (#23306)

mbfrahry committed:
Update CHANGELOG.md for #23306

tombuildsstuff committed:
Merge pull request #23320 from TheisFerre/update-mysql-flexible-server-sdk

changed sdk version to 2022-01-01 for mysql flexible server

tombuildsstuff committed:
updating to include #23320

tombuildsstuff committed:
removing #23306 since it's not user facing / fmt

wuxu92 committed:
add missing properties in webapplication firewall policy (#23239)

mbfrahry committed:
Fix remaining issues coming from make lint (#23330)

bianyifan committed:
`azurerm_data_factory_dataset_parquet` - Support `azure_blob_fs_location` (#23261)

* `data_factory_dataset_parquet` support `azure_blob_fs_location`

* support read azure_blob_fs_location

* add acc test

* ExactlyOneOf and doc

* also update azure_blob_fs_location for dataset_delimited_text

lonegunmanb committed:
add mongodb connection string for azurerm_cosmosdb_account resource (#23331)

bianyifan committed:
New resource: `azurerm_data_factory_dataset_azure_sql_table` (#23264)

* try add azurerm_data_factory_dataset_azure_sql_table

* try acc test

* make it a typed resource

* fix acc test

* try use schema

* minor update

* update schema column type

* change linked_service_name to linked_service_id

* doc

* update

* requiresImport test

teowa committed:
`azurerm_app_configuration` - support for `replica` (#22452)

* app configuration support replica

* fix

* fix datasource doc

stephybun committed:
Update for #23239 #23261 #23331 #22452 #23264

stephybun committed:
Update for #23161 #23302 #23262 #23310 #23263 #23108

stephybun committed:
Update for #23011 #23277 #23260  #22833 #23288 #23147 #22979

tombuildsstuff committed:
dependencies: updating to `v0.20230918.1115907` of `github.com/hashicorp/go-azure-sdk`

xiaxyi committed:
`azurerm_linux_function_app`, `azurerm_windows_function_app` - set the storage account for function app using elastic plans automatically (#21212)

jackofallops committed:
Update CHANGELOG.md for #21212

tombuildsstuff committed:
Merge pull request #23337 from hashicorp/dependencies/go-azure-sdk

dependencies: updating to `v0.20230918.1115907` of `github.com/hashicorp/go-azure-sdk`

tombuildsstuff committed:
updating to include #23337

mbfrahry committed:
`azurerm_key_vault_key` - ForceNew when `expiration_date` is removed from the config file (#23327)

mbfrahry committed:
Update CHANGELOG.md for #23327

teowa committed:
`azurerm_management_group_subscription_association` - correctly mark as gone if not found during read (#23335)

* fix read

* fix

ziyeqf committed:
`azurerm_log_analytics_solution`: fix update & refactor to typed sdk (#23333)

* `azurerm_log_analytics_solution`: add a test for update

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

* fix #23312

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

* `azurerm_log_analytics_solution`: refactor to typed sdk

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

* add state migration

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

* determine if `tag` has changes

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

---------

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

magodo committed:
`azurerm_container_group` - Supports CMK with user assigned identity (#23332)

* `azurerm_container_group` - Supports CMK with user assigned identity

* Renaming & minor clean up

ms-henglu committed:
`azurerm_data_factory` - support for the `publish_enabled` property (#23344)

* `azurerm_data_factory` - support for the `publish_enabled` property

* rename to `publishing_enabled`

AnthonyStainer committed:
`azurerm_container_app` - Correct commands issue with multiple containers (#23338)

Co-authored-by: jackofallops <11830746+jackofallops@users.noreply.github.com>
Co-authored-by: Anthony Stainer <anthony.stainer@egress.com>

jackofallops committed:
Update CHANGELOG.md for #23338

teowa committed:
`azurerm_monitor_diagnostic_setting` - exactly one of  `category` or `category_group` must be specified in `enabled_log` and `log` block (#23308)

* add category category_group validation for non-empty string

* add Exactly one of category or category_group must be specified

* fix lint: change if-else to switch

Sewci0 committed:
`azurerm_storage_account_customer_managed_key` - support for cross-tenant customer-managed keys (#20356)

* Implement federated_identity_client_id && key_vault_uri within SA CMK resource

* Fix required with

* Add acceptance tests

* Fix docs

* Fix tests

* Ensure right permission

* Format code

* Fix linting

* Update website/docs/r/storage_account_customer_managed_key.html.markdown

Co-authored-by: Tom Bamford <tom@bamford.io>

* azurerm_storage_account_customer_managed_key: fix up test config (second set of creds not needed)

* TeamCity configuration for secondary Azure tenant and subscription

---------

Co-authored-by: Tom Bamford <tom@bamford.io>

manicminer committed:
Changelog for #20356

stephybun committed:
Update for #23335 #23332 #23333 #23308

wuxu92 committed:
`azurerm_key_vault_certificate`: fix `createCertificate` lost `http.Response` which is used to check if should recover this certificate (#23204)

* fix recover certificate, update error log and update acctest for recovery certificate

* fix typo

* lint comment fix

stephybun committed:
Update for #23204

WodansSon committed:
including #23217 fixes #23157

* Initial Check-in...

* Fix up test cases...

* Wrap behavior change in feature flag...

* Aligning Synapse test case RG names...

* Update doc and error msg...

* Minor doc fix...

* Refactor code for new behavior...

* Fix lint issues...

* Fix test lint errors...

* Address PR comments...

* Address PR comments...

WodansSon committed:
updating to include #23217

WodansSon committed:
updating note for #23217

katbyte committed:
Update CHANGELOG.md

katbyte committed:
v3.74.0

magodo committed:
`azurerm_management_lock` - Add post-create/delete polling to tolerate RP propagation (#23345)

* `azurerm_management_lock` - Add post-create/delete polling to tolerate RP propagation

* Using ctx deadline

wuxu92 committed:
add description for nat/network rules in firewall policy rule (#23354)

fdmsantos committed:
Add Support for Icon URL in azurerm_bot_service_azure_bot resource (#23114)

* Add Support for Icon URL in azurerm_bot_service_azure_bot resource

* Apply suggestions from code review

Co-authored-by: Tom Harvey <tombuildsstuff@users.noreply.github.com>

* Add icon_url attribute to documentation

* update property description

* Add icon_url attribute to acceptance test

* Add default value to icon_url parameter

Co-authored-by: stephybun <steph@hashicorp.com>

* rebase and terrafmt

* switch to pointer.From in read

---------

Co-authored-by: Tom Harvey <tombuildsstuff@users.noreply.github.com>
Co-authored-by: stephybun <steph@hashicorp.com>

magodo committed:
Fix 2  incorrect import specs (#23368)

tombuildsstuff committed:
r/private_endpoint: normalizing the value of a Kusto Cluster ID (#23357)

Fixes #23356

teowa committed:
new resource `azurerm_resource_management_private_link` (#23098)

* new resource azurerm_resource_management_private_link

* fix review comment

* fix doc

* fix test

ziyeqf committed:
`azurerm_log_analytics_workspace`: support `data_collection_rule_id` (#23347)

* add `default_data_collection_rule_id`

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

* add doc

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

* update per comments

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

* update per comments

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

* update per comments

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

---------

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

muppet2011ad committed:
Docs: Clarify meaning of `external_enabled = true` for azurerm_container_app ingress (#23360)

* Clarify what external ingress means for aca

* New wording for external_enabled

Co-authored-by: jackofallops <11830746+jackofallops@users.noreply.github.com>

---------

Co-authored-by: jackofallops <11830746+jackofallops@users.noreply.github.com>

jackofallops committed:
`paloalto` - Fix rulestack commit on delete for sub resources (#23362)

jackofallops committed:
Update CHANGELOG.md for #23362

sinbai committed:
`azurerm_api_management_api` : fix importing `openapi` format content file issue (#23348)

mbfrahry committed:
Update CHANGELOG.md for #23348

mbfrahry committed:
dependencies: migrate `firewall` to `hashicorp/go-azure-sdk` (#22863)

* Swap various azurerm_firewal_x resources to hashicorp/go-azure-sdk

* move azurerm_firewall_policy to hashicorp/go-azure-sdk

* move azurerm_firewall_policy_rule_collection_group to hashicorp/go-azure-sdk

* Hardcode Dnat for sdk inconsistencies

* Only set identity when type is not None

* Add deadline for disappears test

* move check for existing virtual hub

* Add deadline to test

ms-henglu committed:
`azurerm_kubernetes_cluster` - `network_profile.network_policy` supports `cilium` (#23342)

* `azurerm_kubernetes_cluster` - `network_profile.network_policy` supports `cilium`

* remove forcenew

liuwuliuyun committed:
`azurerm_cognitive_deployment` - remove ForceNew tag for property 'capacity' (#23251)

* remove ForceNew tag

* add update function

* format update

myc2h6o committed:
`data` `azurerm_shared_image_version` - consider `excludeFromLatest` for `latest` version (#21822)

Co-authored-by: kt <kt@katbyte.me>

Ihab-Zhaika committed:
security_center_subscription_pricing_resource: refactor disabled extensions logic (#22997)

* security_center_subscription_pricing_resource: Fix disabled extensions logic

Signed-off-by: Ihab Zhaika <ihabzhaika@microsoft.com>

* Add backend extensions explicitly

Signed-off-by: Ihab Zhaika <ihabzhaika@microsoft.com>

* Support extension per subplan

Signed-off-by: Ihab Zhaika <ihabzhaika@microsoft.com>

* Update the documentation

Signed-off-by: Ihab Zhaika <ihabzhaika@microsoft.com>

* Add tests

Signed-off-by: Ihab Zhaika <ihabzhaika@microsoft.com>

* Add containers bundle extnesions

Signed-off-by: Ihab Zhaika <ihabzhaika@microsoft.com>

* remove hardocded extensions and fetch them dynamiclly

Signed-off-by: Ihab Zhaika <ihabzhaika@microsoft.com>

* Fail when switching to free and having extensions

Signed-off-by: Ihab Zhaika <ihabzhaika@microsoft.com>

* Apply suggestions from code review

review changes

Co-authored-by: catriona-m <86247157+catriona-m@users.noreply.github.com>

---------

Signed-off-by: Ihab Zhaika <ihabzhaika@microsoft.com>
Co-authored-by: catriona-m <86247157+catriona-m@users.noreply.github.com>

catriona-m committed:
Update CHANGELOG.md #22997

unique-dominik committed:
Doc Fix: Update pim_eligible_role_assignment.html.markdown by removing `principal_type` as required argument.(#23389)

TheisFerre committed:
Mysql flexible server iops scaling (#23329)

* add support for iops scaling

* add validation to create function

* add documentation for io_scaling_enabled

* add test with io_scaling enabled

* formatted test string

* Changed iops to upper case for documentation

* added test config with no iops

* set deault of io_scaling_enabled to false

* run terrafmt to format config strings

* change default value to 'false'

ziyeqf committed:
update test (#23299)

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

ziyeqf committed:
New Resource: `azurerm_application_load_balancer` (#22517)

jackofallops committed:
Update CHANGELOG.md for #22517

WodansSon committed:
including #23384 fixes #23376

* Initial Check-in...

* Fix error message grammar...

* Fix the same error message again...

* Initial Check-in...

WodansSon committed:
updating to include #23384

ziyeqf committed:
New Resource: `azurerm_application_load_balancer_frontend`

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

jackofallops committed:
`servicenetworking` - fixup test checks (#23410)

stephybun committed:
Update for #23345 #23204 #23344 #23308 #23333 #23332 #23335 #23354 #23114 #23098 #23342 #23329 #23251

WodansSon committed:
updating to include #23419

* Initial Check-in...

* Change default sku_name value for 4.0...

* Fix lint error...

* Update 3.x to be computed...

* Make Sku conditional in Create function...

WodansSon committed:
Update CHANGELOG.md to include #23419

katbyte committed:
v3.75.0

mbfrahry committed:
`azurerm_static_site` - Add support for `app_settings` (#23421)

mbfrahry committed:
Update CHANGELOG.md for #23421

vszabo2 committed:
Update frontdoor.html.markdown (#23429)

https://github.com/hashicorp/terraform-provider-azurerm/blob/da7468226f353d7933e09686604309899ff3a528/internal/services/frontdoor/frontdoor_resource.go#L1989-L1990

ms-henglu committed:
new resource: `azurerm_spring_cloud_application_insights_application_performance_monitoring` (#23107)

katbyte committed:
Update CHANGELOG.md #23107

chouse-qumodity committed:
Update kubernetes_cluster.html.markdown fix exports/supports

`allowed` and `not_allowed` were described as "block exports the following" when they should say "block supports the following"

jackofallops committed:
`TypedSDK` - add filtering tag for removed schema items (#23415)

wuxu92 committed:
use normalize location for local rulestack resource

tombuildsstuff committed:
New Tool: `update-go-azure-sdk`

This tool is intended to update the `hashicorp/go-azure-sdk` dependency within the Provider - and then update the codebase
to take advantage of any new API Versions that are present within this version of `hashicorp/go-azure-sdk`, if used in the
Provider and the API versions are compatible (e.g. the tests pass when updating the API Version).

This tool makes a considerable number of assumptions about the working directory and environment this is being run in and
is primarily intended to be run in automation - and whilst will not be automatically configured initially, in time should
mean that both SDK upgrades and new API Version upgrades are (mostly) automated.

In the event that an SDK/API Version is _not_ compatible, the automated process will fail - and manual intervention is
required - which is desired (since this should be the edge-case and/or requires additional thought).

hashicorp-tsccr[bot] committed:
Result of tsccr-helper -log-level=info -pin-all-workflows .

tombuildsstuff committed:
Merge pull request #23484 from hashicorp/tsccr-auto-pinning/trusted/2023-10-09

SEC-090: Automated trusted workflow pinning (2023-10-09)

manicminer committed:
Merge pull request #23444 from chouse-qumodity/patch-1

Update kubernetes_cluster.html.markdown fix exports/supports

neil-yechenwei committed:
azurerm_cosmosdb_postgresql_cluster - update the validation of sql_version and citus_version (#23476)

* azurerm_cosmosdb_postgresql_cluster - update the validation of sql_version and citus_version

* update code

manicminer committed:
Changelog for #23476

ergihu committed:
Update nat_gateway.html.markdown to add ip association examples

manicminer committed:
Merge pull request #23483 from wuxu92/paloato/loc

`azurerm_palo_alto_local_rulestack`: use `location.Normalize` in Create method

manicminer committed:
Changelog for #23483

manicminer committed:
Update nat_gateway.html.markdown to link to nat_gateway_public_ip_association and nat_gateway_public_ip_prefix_association resources

manicminer committed:
Merge pull request #23502 from ergihu/r/azurerm_nat_gateway_docs_ip_association

Update nat_gateway.html.markdown to add ip association examples

tombuildsstuff committed:
tools/update-go-azure-sdk: `output-file` is now optional and `old-sdk-version` is now auto-detected

tombuildsstuff committed:
tools/update-go-azure-sdk: `--working-directory` is now `--azurerm-repo-path`

tombuildsstuff committed:
tools/update-go-azure-sdk: support for `--go-sdk-repo-path`

This allows for reusing an existing checkout, rather than cloning a fresh copy of `hashicorp/go-azure-sdk`
useful both for automation, and when working with an offline copy

tombuildsstuff committed:
tools/update-go-azure-sdk: outputting a message when no services were updated

tombuildsstuff committed:
tools/update-go-azure-sdk: updating the readme example/fmt

tombuildsstuff committed:
go mod tidy

tombuildsstuff committed:
tools/update-go-azure-sdk: updating the commit logic

tombuildsstuff committed:
tools/update-go-azure-sdk: updating the import search to use asts

catriona-m committed:
`azurerm_eventhub` - remove ForceNew and check `partition_count` is not decreased (#23499)

* remove ForceNew and check partition_count is not decreased

* update doc

catriona-m committed:
Update CHANGELOG.md #23499

tombuildsstuff committed:
dependencies: updating to Go 1.21.3

tombuildsstuff committed:
go mod tidy

tombuildsstuff committed:
Merge pull request #23498 from hashicorp/tool/update-go-azure-sdk

New Tool: `update-go-azure-sdk`

tombuildsstuff committed:
Merge pull request #23514 from hashicorp/f/updating-to-go-1.21.3

dependencies: updating to Go 1.21.3

tombuildsstuff committed:
updating to include #23514

ziyeqf committed:
update per comments

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

ziyeqf committed:
file renaming

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

ziyeqf committed:
re-trigger ci

Signed-off-by: ziyeqf <51212351+ziyeqf@users.noreply.github.com>

katbyte committed:
Update internal/services/servicenetworking/application_load_balancer_frontend_resource.go

Co-authored-by: catriona-m <86247157+catriona-m@users.noreply.github.com>

katbyte committed:
Update internal/services/servicenetworking/application_load_balancer_frontend_resource.go

Co-authored-by: catriona-m <86247157+catriona-m@users.noreply.github.com>
